lib/service/checkpayments.go

```diff
@@ -42,10 +44,29 @@ func (svc *LndhubService) CheckAllPendingOutgoingPayments(ctx context.Context) (
 }
 
 func (svc *LndhubService) GetTransactionEntryByInvoiceId(ctx context.Context, id int64) (models.TransactionEntry, error) {
-    entry := models.TransactionEntry{}
+	entry := models.TransactionEntry{}
+	feeReserveEntry := models.TransactionEntry{}
 
-    err := svc.DB.NewSelect().Model(&entry).Where("invoice_id = ?", id).Limit(1).Scan(ctx)
-    return entry, err
+	err := svc.DB.NewSelect().Model(&entry).Where("invoice_id = ? and entry_type = ?", id, models.EntryTypeOutgoing).Limit(1).Scan(ctx)
+	if err != nil {
+		//migration issue: pre-feereserve payment will cause a "no rows in result set" error.
+		//in this case, we also look for the entries without the outgoing check, and do not add the fee reserve
+		//we can remove this later when all relevant payments will have an entry_type and a fee_reserve tx
```

Comment:
I think we can get rid of it after [2 weeks](https://github.com/getAlby/lndhub.go/blob/main/lib/service/checkpayments.go#L19) because we won't encounter any more pre-feereserve payments in this code block by then.

Reply:
it's a generic `GetTransactionEntryByInvoiceId` method so it should be independent of where it is used. 
if we don't migrate the data this function will be broken when used elsewhere.